cmd/client-s3.go

```diff
@@ -762,6 +763,11 @@ func (c *s3Client) MakeBucket(region string) *probe.Error {
 	}
 	e := c.api.MakeBucket(bucket, region)
 	if e != nil {
+		// Enhance error message when invalid region error is received
+		if errResp, ok := e.(minio.ErrorResponse); ok && errResp.Code == "InvalidRegion" {
+			errResp.Message = fmt.Sprintf("The server rejected `%s` region. Kindly pass the correct region value using `--region` flag.", region)
```

Comment:
We actually retry with the Region returned in the headers. not sure if this is needed..